pkgs/misc/vim-plugins/default.nix

```diff
@@ -429,6 +429,20 @@ rec {
 
   };
 
+  vim-powerline = buildVimPluginFrom2Nix { # created by nix#NixDerivation
+    name = "vim-powerline-2012-11-23";
+    src = fetchgit {
+      url = "https://github.com/Lokaltog/vim-powerline";
+      rev = "09c0cea859a2e0989eea740655b35976d951a84e";
+      sha256 = "1anndrspxciy53xpzc1sn6ydszij3riw5ijamadvw34rl30vffk6";
+    };
+    dependencies = [];
+    preFixup = ''
+      sed "s/\(let\ g:Powerline_cache_dir\ =\).*/\1\ \/var\/cache\//" \
+        -i "$out"/share/vim-plugins/vim-powerline/plugin/Powerline.vim
```

Comment:
`/var/cache` should not be writeable by users. Something in `$HOME` is preferred here.

Also the following should look more readable:

```
sed 's!\(let g:Powerline_cache_dir =\).*!\1/var/cache/!' \
        -i "$out"/share/vim-plugins/vim-powerline/plugin/Powerline.vim
```

Reply:
I want to use `$HOME/.vim` as `Powerline_cache_dir` .  To make sure it exist i tried to use this code 
```bash
if [ ! -d "$HOME/.vim" ]; then
      mkdir $HOME/.vim
fi
```
in `PreFixup`
nix-sell gives the error:
` mkdir: cannot create directory '/homeless-shelter/.vim': No such file or directory`

Reply:
This must happen at run-time (when the plugins is actually loaded) and not at build-time (in the nix-build sandbox).

In vim script this could be achieved using the following code,

```
silent !mkdir -p ~/.vim/powerline.cache > /dev/null 2>&1
```

but probably it is better to ask upstream first to implement this.